src/toil/wdl/wdltoil.py

```diff
@@ -5500,6 +5501,29 @@ def main() -> None:
                         )
                         options.all_call_outputs = True
 
+                WDL.Lint.lint(document)
+
+                strict = options.strict
+                shown = [0]
+
+                fp = io.StringIO()
+                outline(
+                    document,
+                    0,
+                    file=fp,
+                    show_called=(document.workflow is not None),
+                    shown=shown,
+                )  # type: ignore[no-untyped-call]
```

Comment:
This section could use some comments to explain how we're using `shown` as a way to get data out of a function by letting it be mutated, and we're using `fp` not just as any old file object but as a way to capture text about lint warnings, and that `outline` despite its generic name is going to count up the lint warnings into `shown` and generate descriptions of them into `fp`. Unless you know why this code is here already, it's not very obvious what it's doing or that it needs to come after the `lint()` call to work properly.

Reply:
`shown` and `fp` are also not great variable names that obviously are lint error related. If they aren't contained in a linting-related function, and there's no section-delimiting comments, and they're just bouncing around `main` with everything else, it might be good to give them names that make their purpose clearer.